# CrunchieMunchies

You work in marketing for a food company <b>myCorps</b>, which is developing a new kind of tasty, wholesome cereal called <b>CrunchieMunchies</b>. 

You want to demonstrate to consumers how healthy your cereal is in comparison to other leading brands, so you’ve dug up nutritional data on several different competitors.

Your task is to use <em>NumPy statistical calculations</em> to analyze this data and prove that your <b>CrunchieMunchies</b> is the healthiest choice for consumers.






# Task STEPS


1.First, import numpy.

In [3]:
import numpy as np

2.Look over the <b><em>cereal.csv</em></b> file. This file contains the reported calorie amounts for different cereal brands. Load the data from the file and save it as <b><em>calorie_stats.</em></b>



In [101]:
calorie_stats = np.genfromtxt('cereal.csv',delimiter=',')
print(calorie_stats)

[ 70. 120.  70.  50. 110. 110. 110. 130.  90.  90. 120. 110. 120. 110.
 110. 110. 100. 110. 110. 110. 100. 110. 100. 100. 110. 110. 100. 120.
 120. 110. 100. 110. 100. 110. 120. 120. 110. 110. 110. 140. 110. 100.
 110. 100. 150. 150. 160. 100. 120. 140.  90. 130. 120. 100.  50.  50.
 100. 100. 120. 100.  90. 110. 110.  80.  90.  90. 110. 110.  90. 110.
 140. 100. 110. 110. 100. 100. 110.]


3.There are <em>60 calories per serving of CrunchieMunchies</em>. How much <b>higher</b> is the <b>average calorie count</b> of your competition?

Save the answer to the variable <b>average_calories</b> and print the variable to the terminal to see the answer.


In [16]:
average_calories = np.average(calorie_stats)
print(average_calories)

106.88311688311688


4.Does the <b>average calorie count</b> adequately reflect the distribution of the dataset? Let’s sort the data and see.

<b><em>Sort</em></b> the data and save the result to the variable <b>calorie_stats_sorted</b>. Print the sorted data to the terminal.


In [14]:
calorie_stats_sorted = np.sort(calorie_stats)
print(calorie_stats_sorted)

[ 50.  50.  50.  70.  70.  80.  90.  90.  90.  90.  90.  90.  90. 100.
 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100.
 100. 100. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110.
 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110.
 110. 110. 110. 120. 120. 120. 120. 120. 120. 120. 120. 120. 120. 130.
 130. 140. 140. 140. 150. 150. 160.]


5.Do you see what I’m seeing? Looks like <b><em>the majority of the cereals are higher than the mean</em></b>. Let’s see if the <b>median</b> is a better representative of the dataset.

Calculate the median of the dataset and save your answer to <b><em >median_calories</em></b>. Print the median so you can see how it compares to the mean.

In [104]:
median_calories = np.median(calorie_stats)
print(median_calories)

110.0


6.While the median demonstrates that <b><em><q>at least half of our values are over 100 calories</q></em></b>, it would be more impressive to show that a significant portion of the competition has a higher calorie count that CrunchieMunchies.

<b>Calculate different percentiles</b> and print them to the terminal until you find the lowest percentile that is greater than 60 calories. Save this value to the variable <b>nth_percentile</b>.


In [106]:
nth_percentile = np.percentile(calorie_stats, 4)
print(nth_percentile)

70.0


7.While the percentile shows us that<b><em><q>the majority of the competition has a much higher calorie count</q></em></b>, it’s an awkward concept to use in marketing materials.

Instead, let’s calculate the percentage of cereals that <b><em><q>have more than 60 calories per serving</q></em></b>. Save your answer to the variable <b><em>more_calories</em></b> and print it to the terminal

In [110]:
total_cereals = np.size(calorie_stats)
cereals_above_60 = np.size(calorie_stats[calorie_stats > 60])
more_calories= (cereals_above_60 / total_cereals) * 100
print("Percentage of cereals above 60 calories:", more_calories, "%")

Percentage of cereals above 60 calories: 96.1038961038961 %


8.Wow! That’s a really high percentage. That’s going to be very useful when we promote CrunchieMunchies. But one question is, <b><em>how much variation exists in the dataset? </b></em></q>Can we make the generalization that most cereals have around 100 calories or is the spread even greater?

Calculate the amount of variation by finding the <b><em>standard deviation</em></b> Save your answer to <b><em>calorie_std</em></b> and print to the terminal. How can we incorporate this value into our analysis?

In [94]:
calorie_std = np.std(calorie_stats)
print(calorie_std)

19.35718533390827


9.Write a short paragraph that sums up your findings and how you think this data could be used to 
<b>myCorp’s</b> advantage when marketing CrunchieMunchies.


<h2>Analysis of my findings</h2>

This data definitely is very advantageable for <b>myCorp’s</b> because the calculations clearly shows that more than
<b>96%</b> of other cereal brands has high calories per serving than Crunchie Munchies and on the average, calories count of other brands is <b>106</b> which reflects that Crunchie Munchies with <b>60 calories/serving</b> are far behind the value 
with the variation of <b>19.35</b> in dataset.

These numbers will going to be very productive to attract and seize customers as well as in marketing, advertisement and other brand promotions. 